tests/helpers/TemporaryFilename.h

```diff
@@ -0,0 +1,78 @@
+/* Copyright (c) 2015-2022 EPFL/Blue Brain Project
+ * All rights reserved. Do not distribute without permission.
+ * Responsible Author: Nadir Roman Guerrero <nadir.romanguerrero@epfl.ch>
+ *
+ * This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#pragma once
+
+#include <filesystem>
+#include <random>
+#include <string>
+
+class TemporaryFilename
+{
+public:
+    static std::string generateFake(size_t len = 10)
+    {
+        std::string result;
+        do
+        {
+            result = _generate(len);
+        } while (_exists(result));
+
+        return "/" + result + "/fake_file";
+    }
+
+    static std::string generateValid(size_t len = 10)
+    {
+        std::string result;
+        do
+        {
+            result = "/tmp/" + _generate(len);
+        } while (_exists(result));
```

Comment:
If you do directly std::filesystem::exists(result) it doesn't work ?

Reply:
I just put it on a separate static function to leave the main entry function of the class cleaner. After inlining, it will result in the same code you are suggesting.